packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin.tsx

```diff
@@ -285,3 +300,27 @@ export function positionTooltip(u: uPlot, bbox: DOMRect) {
 
   return { x, y };
 }
+
+/**
+ * Given a sort order and a value, return a function that can be used to sort values
+ * Null values are always sorted to the end regardless of the sort order
+ */
+function sortValues(sort: TooltipSortOrder) {
+  return (a: number | null | undefined, b: number | null | undefined) => {
+    if (a === b) {
+      return 0;
+    }
+
+    if (b == null) {
+      return -1;
+    }
+    if (a == null) {
+      return 1;
+    }
+
+    if (sort === TooltipSortOrder.Descending) {
+      return b - a;
+    }
+    return a - b;
```

Comment:
```suggestion
    const order = a > b ? 1: -1;
    if (sort === TooltipSortOrder.Descending) {
      return order * -1;
    }
    return order;
```

Can we implement this so it also works for string inputs?  Maybe something like the above (untested!!!) approach.  For the time series tooltip it likely makes no difference, but if forced to numbers, then we will for sure need custom implementations for other panels (though we are already working on that anyway)

Reply:
you'd want to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

if you want it to be fast, you'd first init & cache an Intl.Collator instance and use that: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#performance

Reply:
That's done now, thanks for the feedback folks!